app/controllers/assessments_controller.rb

```diff
@@ -211,16 +211,19 @@ def update
           delta << "#{symptom.label} (\"#{old_val}\" to \"#{new_val}\")"
         end
       end
-
       assessment.symptomatic = assessment.symptomatic?
+      old_reporter = assessment.who_reported
       # Monitorees can't edit their own assessments, so the last person to touch this assessment was current_user
       assessment.who_reported = current_user.email
 
       # Attempt to save and continue; else if failed redirect to index
       return unless assessment.save
 
       comment = 'User updated an existing report (ID: ' + assessment.id.to_s + ').'
-      comment += ' Symptom updates: ' + delta.join(', ') + '.' unless delta.empty?
+      unless delta.empty?
+        comment += ' Symptom updates: ' + delta.join(', ') + '.'
+        comment += " Reporter updated: (\"#{old_reporter}\" to \"#{current_user.email}\")." unless old_reporter == current_user.email
```

Comment:
Could we add on to the update test helper in assessments_controller_test.rb to validate the history item?

```ruby
assert_histories_contain(assessment.patient, "Reporter updated: (\"Monitoree\" to \"#{user.email}\")")
```

It could potenitally go further by updating the assessment twice and testing the `unless old_reporter == current_user.email` logic

Reply:
@jkufro how important do you think updating it twice and testing the unless logic is? It's not particularly difficult to add but it does make the code for the tests themselves less clean since you have to loop through the same code twice but make a different assertion after each run.

Reply:
I think this would work and keeps it fairly clean
```ruby
  def success_update_report_test(user)
    patient_submission_token = patients(:patient_1).submission_token
    assessment = assessments(:patient_1_assessment_2)

    # edit the assessment
    symptoms = symptoms_param(assessment.reported_condition.symptoms)
    expected_change = !symptoms.first[:value]
    symptoms.first[:value] = expected_change
    sign_in user
    assert_changes 'History.count' do
      assert_no_difference 'AssessmentReceipt.count' do
        assert_no_difference 'Assessment.count' do
          post :update, params: {
            patient_submission_token: patient_submission_token,
            id: assessment.id,
            experiencing_symptoms: 'yes',
            symptoms: symptoms
          }
          assert_equal 204, response.code.to_i
        end
      end
      assessment.reload
      assert_equal expected_change, assessment.reported_condition.symptoms.first.bool_value
    end
  end

  test 'successfully update report as public_health_user' do
    success_update_report_test(create('public_health_user'))
  end

  test 'successfully update report as public_health_enroller_user' do
    success_update_report_test(create('public_health_enroller_user'))
  end

  test 'successfully update report as contact_tracer_user' do
    success_update_report_test(create('contact_tracer_user'))
  end

  test 'successfully update report as super_user' do
    success_update_report_test(create('super_user'))
  end

  test 'reported updated report history item' do
    ph_user = create('public_health_user')
    ct_user = create('contact_tracer_user')

    # Edit report and expect a specific snippet of a history item
    success_update_report_test(ph_user)
    assert_histories_contain(patients(:patient_1), "Reporter updated: (\"Monitoree\" to \"#{ph_user.email}\")")

    # Delete all to ensure that the history is not created again
    patients(:patient_1).histories.delete_all

    # Edit report and expect no history item because the report has not changed
    success_update_report_test(ct_user)
    assert_not_histories_contain(patients(:patient_1), "Reporter updated: (\"Monitoree\" to \"#{ct_user.email}\")")
  end
```

Reply:
@jkufro done!